### PR TITLE
fix: update sentence_encoder to jina 3

### LINKED ISSUE #319: Refactoring of DocumentArrayInMemory affects TransformerSentenceEncoder
Updates on DocumentArrayInMemory where `get_attributes` has become private access in [this update](https://github.com/jina-ai/docarray/blob/97866c6dc27602364f76d7e582c3c26be40f7b99/CHANGELOG.md#-code-refactoring) which has made TransformerSentenceEncoder outdated as its class is accessing [`get_attributes`](https://github.com/jina-ai/executors/blob/c9bffccc8f97dc33128c48db7a0f69336448008e/jinahub/encoders/text/TransformerSentenceEncoder/sentence_encoder.py#L57)

The error thrown is:
```
AttributeError: 'DocumentArrayInMemory' object has no attribute 'get_attributes'
```

### PR DESCRIPTION
**Pull request by nan-wang**
Goals:

- update sentence_encoder to jina 3

resolved #319 

PR feedback: @jina-ai/workstream-hub

### CODE REVIEW

File: jinahub/encoders/text/TransformerSentenceEncoder/sentence_encoder.py [hunk 1]
```diff
@@ -45,18 +43,14 @@ def encode(
...
-        ).batch(
-            batch_size=parameters.get('batch_size', self.batch_size),
-        ):
-            texts = batch.get_attributes('text')
-
-            with torch.inference_mode():
-                embeddings = self.model.encode(texts)
-                for doc, embedding in zip(batch, embeddings):
-                    doc.embedding = embedding
+        document_batches_generator = DocumentArray(
```
Comment (JoanFM) [inline jinahub/encoders/text/TransformerSentenceEncoder/sentence_encoder.py:46]: I think is iterator not generator

File: jinahub/encoders/text/TransformerSentenceEncoder/tests/unit/test_executor.py [hunk 2]
```diff
@@ -80,7 +80,7 @@ def test_traversal_path(
 
     basic_encoder.encode(docs=docs, parameters={'traversal_paths': traversal_paths})
     for path, count in counts:
-        embeddings = docs.traverse_flat([path]).get_attributes('embedding')
+        embeddings = DocumentArray(docs.traverse_flat(path)).embeddings
```
Comment (JoanFM) [inline jinahub/encoders/text/TransformerSentenceEncoder/tests/unit/test_executor.py:83]: traverse_flat is not good, u should do `docs[path]`

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: jinahub/encoders/text/TransformerSentenceEncoder/sentence_encoder.py (hunk 1 above)
Comment (nan-wang) [reply jinahub/encoders/text/TransformerSentenceEncoder/sentence_encoder.py:46]: it is a generator
```python
da = DocumentArray.empty(10)
it=da.batch(batch_size=3)
type(it)
Out[5]: generator
```

File: jinahub/encoders/text/TransformerSentenceEncoder/tests/unit/test_executor.py (hunk 2 above)
Comment (nan-wang) [reply jinahub/encoders/text/TransformerSentenceEncoder/tests/unit/test_executor.py:83]: yes, you r right. I missed this part

**Comment by JoanFM** [comment]
I agree with @JohannesMessner

**Comment by samsja** [comment]
here is the new repo https://github.com/jina-ai/executor-text-transformers-sentence-encoder

**Comment by nan-wang** [comment]
released the new executor at https://github.com/jina-ai/executor-sentence-transformer/releases/tag/v0.5
One can find the executor at https://hub.jina.ai/executor/nh1ucuki

**Pull request closed without merging**